File: src/2006.count-number-of-pairs-with-absolute-difference-k.2116/solution.ts
function countKDifference (nums: number[], k: number): number {
    const map = new Map<number, number>();
    let result = 0;
    for (let i = 0; i < nums.length; i++) {
        if (map.has(nums[i] + k)) {
            result += map.get(nums[i] + k);
        }
        if (map.has(nums[i] - k)) {
            result += map.get(nums[i] - k);
        }
        map.set(nums[i], map.has(nums[i]) ? map.get(nums[i]) + 1 : 1);
    }

    return result;
}

File: src/2006.count-number-of-pairs-with-absolute-difference-k.2116/solution.java
class Solution {
    public int countKDifference(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int result = 0;
        for (int n : nums) {
            if (map.containsKey(n + k)) {
                result += map.get(n + k);
            }
            if (map.containsKey(n - k)) {
                result += map.get(n - k);
            }
            map.merge(n, 1, Integer::sum);
        }
        return result;
    }
}